elf_reader.go

```diff
@@ -541,28 +579,67 @@ func mapSpecFromBTF(spec *btf.Spec, name string) (*MapSpec, error) {
 			}
 
 		case "pinning":
+			if inner {
+				return nil, errors.New("inner maps can't be pinned")
```

Comment:
I think it's better to check this at the loading stage in `newMapWithOptions`.

Reply:
Would there be a benefit to having both the ELF reader and `newMapWithOptions` check for this? Created https://github.com/cilium/ebpf/issues/237 to make sure we follow up.